target/linux/ramips/image/mt7621.mk

```diff
@@ -529,6 +529,13 @@ define Device/gnubee_gb-pc2
 endef
 TARGET_DEVICES += gnubee_gb-pc2
 
+define Device/hilink_hlk-7621a
```

Comment:
Device/dsa-migration needs to be added like for the other devices.

Reply:
> Device/dsa-migration needs to be added like for the other devices.

@adschm Excuse me, I don't know how to write the dsa-migration. I referenced this. https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/image/mt76x8.mk#L131

Reply:
This is specific to the mt7621 subtarget in ramips. Since devices have been changed from swconfig to DSA without migration, we have introduced a compatibility version in order to force user to reset their config during upgrade.

For simplicity and some other reason, we set the compat-version to 1.1 for all devices on mt7621:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/mt7621/base-files/etc/board.d/05_compat-version

This means that new devices start with 1.1, and not with the default 1.0. To also support this on the image side, you have to add the relevant include. This will require no additional changes to your PR, but without it sysupgrade will not work.
As an additional benefit, the same mechanism will work if somebody backports the device to 19.07, even if it is done in a local repo, where the device would still require swconfig ...

Reply:
> This is specific to the mt7621 subtarget in ramips. Since devices have been changed from swconfig to DSA without migration, we have introduced a compatibility version in order to force user to reset their config during upgrade.
> 
> For simplicity and some other reason, we set the compat-version to 1.1 for all devices on mt7621:
> https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/mt7621/base-files/etc/board.d/05_compat-version
> 
> This means that new devices start with 1.1, and not with the default 1.0. To also support this on the image side, you have to add the relevant include. This will require no additional changes to your PR, but without it sysupgrade will not work.
> As an additional benefit, the same mechanism will work if somebody backports the device to 19.07, even if it is done in a local repo, where the device would still require swconfig ...

@adschm Thank you! Now I know why the module can't boot if flash sysupgrade firmware(a branch of lede, it doesn't have this feature) at first.😅

Reply:
@adschm I have added this, but it was not detected. Maybe beacause it's the next line😂

Reply:
> Device/dsa-migration needs to be added like for the other devices.

@adschm I've added the Device/dsa-migration, why it still needs changes?
https://github.com/cyijun/openwrt/blob/master/target/linux/ramips/image/mt7621.mk#L533